### PR TITLE
Update eslint and dependencies

### PR DESCRIPTION
**Pull request by joefitter**
- updated eslint to 2.12.0
- updated removed rules
- updated eslint-plugin-filenames to 1.0.0
- updated eslint-plugin-mocha to 3.0.0

### CODE REVIEW

File: rules/stylistic-issues.js [hunk 1]
```diff
@@ -21,28 +21,28 @@ module.exports = {
...
     'no-nested-ternary': 2,
     'no-new-object': 2,
-    'no-space-before-semi': 2,
+    'semi-spacing': 2,
     'no-spaced-func': 2,
     'no-ternary': 0,
     'no-trailing-spaces': 2,
     'no-underscore-dangle': 2,
-    'no-wrap-func': 2,
+    'no-extra-parens': 2,
```
Comment (joefitter) [inline rules/stylistic-issues.js]: @easternbloc this is the rule that changed - we can configure it to only no-wrap-function as before

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @merlinc @noels mind having a 👀  over this?

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @merlinc @noels bump!

**Comment by merlinc** [comment]
@joefitter Oops, this got lost in the ether. Looking now :)

**Comment by merlinc** [comment]
So this upgrades `eslint` to v2, which is great and long overdue!

Are there many new rules - this is just looking like syntactic renames - I'd expected a few new rules to need to be defined.

**Comment by joefitter** [comment]
It looks as though more has been removed/renamed than actually added. There is some handy es6 stuff though like prefer const over let etc - we should use these but out of scope of this PR I guess

**Comment by joefitter** [comment]
I still stand by binning this in favour of https://github.com/feross/standard

**Comment by joefitter** [comment]
although @easternbloc may still need a lot of convincing. The fact that this repo even exists justifies binning it for a sensible set of defaults that never have to be maintained IMHO. Ima keep plugging it until somebody listens ;)

**Comment by merlinc** [comment]
@joefitter Yeah - @chrisns convinced me that each project should choose existing standards that match the style of thing it's doing. `Angular` projects should follow their own eco-system standards and not be forced to comply to `expressjs` standards (for example).

At _best_ I think this project could be a `eslint-config-HOF`, but even then I'm still dubious about it.

`Standard` is better than nothing, but IMHO `eslint` does proper checking for language syntax bugs, together with understanding that test and source directories have different requirements for testing.

**Comment by joefitter** [comment]
Standard is built on top of eslint and uses sensible defaults, including test I believe. The main point being the end developer doesn't have to worry about configuring and maintaining, updating etc, a .eslintrc file and just concentrates on developing the app

**Comment by daniel-ac-martin** [comment]
This LGTM but I'm not an expert on eslint.
Would be nice to get this upgraded.
@easternbloc, @JoeChapman: Any thoughts?

**Comment by easternbloc** [comment]
LGTM

**Pull request merged**